src/index.ts

```diff
@@ -0,0 +1,130 @@
+import * as path from "path"
+import * as fs from "@ts-common/fs"
+import * as md from "@ts-common/commonmark-to-markdown"
+import * as azureMd from "@azure/openapi-markdown"
+import * as ai from "@ts-common/async-iterator"
+import * as jp from "@ts-common/json-parser"
+import * as it from "@ts-common/iterator"
+import * as json from "@ts-common/json"
+import * as sm from "@ts-common/string-map"
+
+export const cli = async <T>(f: (path: string) => AsyncIterable<T>): Promise<number> => {
+  try {
+    const errors = await f("./")
+    let errorsNumber = 0
+    for await (const e of errors) {
+      console.error(e)
+      ++errorsNumber
+    }
+    console.log(`errors: ${errorsNumber}`)
+    return errorsNumber === 0 ? 0 : 1
+  } catch (e) {
+    console.error("INTERNAL ERROR")
+    console.error(e);
+    return 1
+  }
+}
+
+export type JsonParseError = {
+  readonly code: "JSON_PARSE",
+  readonly error: jp.ParseError
+}
+
+export type FileError = {
+  code: "NO_OPEN_API_FILE_FOUND" | "UNREFERENCED_OPEN_API_FILE"
+  readonly message: string
+  readonly readMeUrl: string
+  readonly openApiUrl: string
+}
+
+export type Error = JsonParseError | FileError
+
+export const avocado = (dir: string): ai.AsyncIterableEx<Error> =>
+  fs.recursiveReaddir(path.resolve(dir))
+    .filter(f => path.basename(f).toLowerCase() === "readme.md")
+    .flatMap(validateReadMe)
+
+type Ref = {
+  readonly url: string
+  readonly pointer: string
+}
+
+const parseRef = (ref: string): Ref => {
```

Comment:
should we take this functions out in a "utils" file?

Reply:
`ts-common/json`, `ts-common/json-parser` or a new one `ts-common/json-ref`

Reply:
Will not fix it in this release.